testsuite/testcases/src/two_traffics_test.rs

```diff
@@ -99,3 +100,41 @@ impl NetworkTest for TwoTrafficsTest {
         <dyn NetworkLoadTest>::run(self, ctx)
     }
 }
+
+pub struct ThreeRegionSimulationTwoTrafficsTest {
+    pub traffic_test: TwoTrafficsTest,
+    pub three_region_simulation_test: ThreeRegionSimulationTest,
+}
+
+impl Test for ThreeRegionSimulationTwoTrafficsTest {
+    fn name(&self) -> &'static str {
+        "three region simulation two traffics test"
+    }
+}
+
+impl NetworkLoadTest for ThreeRegionSimulationTwoTrafficsTest {
+    fn setup(&self, ctx: &mut NetworkContext) -> Result<LoadDestination> {
+        self.traffic_test.setup(ctx).unwrap();
+        self.three_region_simulation_test.setup(ctx).unwrap();
+
+        Ok(LoadDestination::FullnodesOtherwiseValidators)
+    }
+
+    fn test(&self, swarm: &mut dyn Swarm, duration: Duration) -> Result<()> {
+        info!(
+            "Running ThreeRegionSimulationTwoTrafficsTest test for duration {}s",
+            duration.as_secs_f32()
+        );
+        self.traffic_test.test(swarm, duration)
+    }
+
+    fn finish(&self, swarm: &mut dyn Swarm) -> Result<()> {
+        self.three_region_simulation_test.finish(swarm)
```

Comment:
I guess traffic_test.finish(..) should also be called for symmetry and in case that ever needs some cleanup?

Reply:
Makes sense.